packages/lodestar/src/chain/blocks/importBlock.ts

```diff
@@ -118,12 +123,36 @@ export async function importBlock(chain: ImportBlockModules, fullyVerifiedBlock:
       throw Error(`merge block parent's parent POW block not found ${powBlock.parentHash}`);
     onBlockPrecachedData.powBlock = powBlock;
     onBlockPrecachedData.powBlockParent = powBlockParent;
-    onBlockPrecachedData.isMergeTransitionBlock = true;
   }
 
   const prevFinalizedEpoch = chain.forkChoice.getFinalizedCheckpoint().epoch;
+  const prevIsMergeTransitionFinalized = chain.forkChoice.getMergeTransitionFinalized();
+
   chain.forkChoice.onBlock(block.message, postState, onBlockPrecachedData);
 
+  // If this was a merge transition block, it has been successfully imported
+  // in forkchoice! Its now PoS all the way!!!!
+  if (onBlockPrecachedData.isMergeTransitionBlock) {
+    const mergeBlock = block.message as bellatrix.BeaconBlock;
+    const mergeBlockHash = toHexString(ssz.bellatrix.BeaconBlock.hashTreeRoot(mergeBlock));
```

Comment:
This is not safe, you must use config to get the block type from the slot. There will be more forks after bellatrix